extension-style/src/main/java/com/mapbox/maps/extension/style/types/PromoteId.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.maps.extension.style.types
+
+import androidx.annotation.Keep
+import com.mapbox.bindgen.Value
+import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource
+import com.mapbox.maps.extension.style.sources.generated.VectorSource
+
+/**
+ * Holds a property type to promote a specific feature for feature state API.
+ *
+ * @param sourceId source layer id of the feature, either source [GeoJsonSource] or [VectorSource].
+ * @param propertyName feature property name.
+ *
+ * For more information see [The online documentation]https://docs.mapbox.com/mapbox-gl-js/style-spec/types/#promoteId).
+ */
+@Keep
+data class PromoteId @JvmOverloads constructor(
+  val sourceId: String? = null,
+  val propertyName: String
+) {
+  internal fun toValue(): Value {
```

Comment:
Is it used in tests only?
if yes - do we need this function at all here? we could add it as extension function in tests or do smth similar

Reply:
it also used in [Builder methods](https://github.com/mapbox/mapbox-maps-android/blob/fd16ddb63b0f065b178fea7fa3b681a925620c70/extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt#L586-L587) of source in order to add the properties to source.